metagraph/src/annotation/representation/column_compressed/annotate_column_compressed.cpp

```diff
@@ -72,6 +73,42 @@ void ColumnCompressed<Label>::add_labels(const std::vector<Index> &indices,
     }
 }
 
+template <typename Label>
+void ColumnCompressed<Label>::add_label_counts(const std::vector<Index> &indices,
+                                               const VLabels &labels,
+                                               const std::vector<uint32_t> &counts) {
+    assert(indices.size() == counts.size());
+
+    const auto &columns = get_matrix().data();
+
+    if (relation_counts_.size() != columns.size())
+        relation_counts_.resize(columns.size());
+
+    for (const auto &label : labels) {
+        const auto j = label_encoder_.insert_and_encode(label);
+
+        if (!relation_counts_[j].size()) {
+            relation_counts_[j] = sdsl::int_vector<>(columns[j]->num_set_bits(), 0, kCountWidth);
+
+        } else if (relation_counts_[j].size() != columns[j]->num_set_bits()) {
+            logger->error("Binary relation matrix was changed while adding relation counts");
+            exit(1);
+        }
+
+        for (size_t i = 0; i < indices.size(); ++i) {
+            if (uint64_t rank = columns[j]->conditional_rank1(indices[i])) {
+                uint64_t count = std::min(counts[i], (uint32_t)sdsl::bits::lo_set[kCountWidth]);
```

Comment:
since `kCountWidth` is fixed, why not store this upper bound as `kMaxCount`?

Reply:
also, why `uint64_t count` when we know that it can be at most 32-bits?

Reply:
done